include/dali/core/tensor_shape.h

```diff
@@ -637,6 +638,32 @@ struct TensorListShapeBase {
     shapes.resize(num_samples * sample_dim);
   }
 
+
+  /**
+   * Append this TensorListShape with shapes from other TensorListShapes.
+   */
+  void concatenate(const std::vector<TensorListShape<sample_ndim>> &tlss) {
```

Comment:
We have `shape_cat` free function that does something different for the TensorShape.

I think this should be named `append` or we should turn it into a free function called `join`.

It probably would be nice to have overloads:
```
TensorListShape join(const TensorListShape &, const TensorListShape&);
TensorListShape join(const ContainerOfTls& lists); // where ContainerOfTls supports size()
```


What about mixing dynamic and static ndim?

Reply:
Went for `append` and making it a member function.

Mixing dynamic & static would be handled in another place, this function assumes, that every `tls` in `tlss` has the same number of dims.